Source/WebKit/UIProcess/WebPageProxy.cpp

```diff
@@ -3319,7 +3319,12 @@ void WebPageProxy::performDragOperation(DragData& dragData, const String& dragSt
     if (!hasRunningProcess())
         return;
 
-    sendWithAsyncReply(Messages::WebPage::PerformDragOperation(dragData, WTFMove(sandboxExtensionHandle), WTFMove(sandboxExtensionsForUpload)), [this, protectedThis = Ref { *this }] (bool handled) {
+    sendToProcessContainingFrame(frameID, Messages::WebPage::PerformDragOperation(frameID, dragData, WTFMove(sandboxExtensionHandle), WTFMove(sandboxExtensionsForUpload)), [this, protectedThis = Ref { *this }, dragData, dragStorageName] (bool handled, std::optional<WebCore::RemoteUserInputEventData> remoteUserInputEventData, std::optional<SandboxExtensionHandle> sandboxExtensionHandle, std::optional<Vector<SandboxExtension::Handle>> sandboxExtensionsForUpload) mutable {
+        if (remoteUserInputEventData) {
+            dragData.setClientPosition(remoteUserInputEventData->transformedPoint);
+            performDragOperation(dragData, dragStorageName, WTFMove(*sandboxExtensionHandle), WTFMove(*sandboxExtensionsForUpload), remoteUserInputEventData->targetFrameID);
+            return;
+        }
```

Comment:
Can we perform the hit test first and then sending the sandbox extensions to the frame based on the test result? It feels a bit risky to send sandbox extension to a process that does not consume it, and forwards extension from a web process to another (what if sender is compromised?).

cc @whsieh who knows drag and drop

Reply:
I thought about this, but it would appear to add an additional round-trip in the most common case (when hitting something handled in the main frame.)